drivers: dp: swdp_bitbang: Update SWD clock calculation

maxd-nordic opened the pull request:
This patch updates the SWD clock calculation to the latest behavior of DAPLink.

I'm doing this patch because I had trouble getting the frequencies right otherwise.

You can find the reference implementation here:
https://github.com/ARMmbed/DAPLink/blob/main/source/daplink/cmsis-dap/DAP.c

maxd-nordic commented:
@fabiobaltieri @henrikbrixandersen @mniestroj @tmon-nordic @jfischer-no Could you have a look?

mniestroj commented on drivers/dp/swdp_ll_pin.h:
```diff
@@ -6,8 +6,9 @@
+#include <hal/nrf_gpio.h>
 
-#if defined(CONFIG_SOC_SERIES_NRF52X)
+#if defined(CONFIG_SOC_SERIES_NRF52X) || defined(CONFIG_SOC_SERIES_NRF53X)
```
This looks like a separate logical change, related only to nRF53, so I'd prefer a distinct commit for that. Anyway, I agree this is needed to make SWDP driver work on nRF53 with default configuration.

fabiobaltieri commented:
@maxd-nordic check the twister errors

```
In file included from /__w/zephyr/zephyr/drivers/dp/swdp_bitbang.c:24:
/__w/zephyr/zephyr/drivers/dp/swdp_ll_pin.h:9:10: fatal error: hal/nrf_gpio.h: No such file or directory
```

jfischer-no commented on drivers/dp/swdp_ll_pin.h:
```diff
@@ -6,6 +6,7 @@
 
 #include <zephyr/kernel.h>
 #include <zephyr/drivers/gpio.h>
+#include <hal/nrf_gpio.h>
```
That should not be necessary, maybe `#include <soc.h>`, only using definition from MDK here.

jfischer-no replied:
The change is still there :-).

maxd-nordic replied:
ah, it was in the wrong commit, fixed it now

maxd-nordic committed:
drivers: dp: swdp_bitbang: hardcode nRF53 SYSCLK

Since CONFIG_SYS_CLOCK_HW_CYCLES_PER_SEC is tied to the slow RTC clock
for the nRF53, use the default SYSCLK of 64MHz instead.

Signed-off-by: Maximilian Deubel <maximilian.deubel@nordicsemi.no>

maxd-nordic committed:
drivers: dp: swdp_bitbang: Update SWD clock calculation

This patch updates the SWD clock calculation to the latest behavior
of DAPLink.

Signed-off-by: Maximilian Deubel <maximilian.deubel@nordicsemi.no>

maxd-nordic commented:
@jfischer-no I regrouped the changes and added back the line before the return. Could you please have another look?
